src/main/java/com/microsoft/jfr/generation/JfrSettingsAsMapClassGenerator.java

```diff
@@ -0,0 +1,61 @@
+package com.microsoft.jfr.generation;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+class JfrSettingsAsMapClassGenerator {
+
+    private static final String MAP_PUT_TEMPLATE = "MAP_PUT";
+    private static final String CLASS_NAME = "CLASS_NAME";
+    private static final String MAP_SUPPLIER_TEMPLATE =
+            "import java.util.HashMap;\n" +
+                    "import java.util.Map;\n" +
+                    "import java.util.function.Supplier;\n" +
+                    "\n" +
+                    "public class "+ CLASS_NAME + " {\n" +
+                    "\n" +
+                    "    public static Supplier<Map<String, String>>  MAP_SUPPLIER = () -> {\n" +
```

Comment:
I'm trying to figure out how you see this being used. I didn't see a test for it. 

You're going to create a new map every time you call the supplier. Is that what you want? Again, I'm missing the use case here.